src/analysis/gMCS/GPR2models.m

```diff
@@ -0,0 +1,259 @@
+            'reactionFormula',[node{child},' -> ',parent],'printLevel',printLevel-1);
+    end
+else
+    model = addReaction(model,strjoin(node,' & '),...
+        'reactionFormula',[strjoin(node,' + '),' -> ',parent],'printLevel',printLevel-1);
+end
+
+
+for i=1:length(head.children)
+    child = head.children(i);
+    % check if the child is final layer
+    if strcmp(class(child),'LiteralNode')
+        gen = genes(str2num(child.id));
+        model = addReaction(model,[gen{1},'_',node{i}],...
+            'metaboliteList',{gen{1},node{i}},'stoichCoeffList',[-1, 1],...
+            'printLevel',printLevel-1);
+    else
+        model = modelParser(model,genes,child,node{i},node{i},printLevel);
+    end
+end
+
+end
+
+function modelOut = reduceModel(model)
+% As the algorithm generates redundant nodes, this function is used to
+% remove these nodes:
+% Ej:       gen -> node -> rule  == gen -> rule
+
+% This is done for every gene of the model
+genes = model.genes;
+for gen=1:length(genes)
+    gen_idx = find(strcmp(model.mets,genes{gen})); % find idx of gene
+    rxn_gen_out = find(model.S(gen_idx,:)<0);   % find reactions that consume gene
+    % for each reaction, find consumption node
+    [node_idx,aux] = find(model.S(:,rxn_gen_out)>0);
+    for i = 1:length(node_idx)
+        % for each reaction, find consumption node
+        node_name = model.mets(node_idx(i));
+        % find reactions out of this node
+        [aux, rxn_idx] = find((model.S(node_idx(i),:)<0));
+        rxn_name = model.rxns(rxn_idx);
+        % in this reaction, change (node -> rule) for (gene -> rule)
+        model = changeRxnMets(model, node_name, genes(gen), rxn_name);   
+    end
+    % remove reactions (gene -> node)
+%     model = removeRxns(model, model.rxns(rxn_gen_out), 'metFlag',false); 
+    % remove stoichimetri of intermediate nodes
+    model.S(node_idx,:)=0;
+    % remove output of genes to intermediate nodes
+    model.S(:,rxn_gen_out)=0;
+end
+
+% remove unused reactions and unused metabolites
+% model = removeTrivialStoichiometry(model);    % problems, removes mets 2 times
+anyRowBool = any(model.S,2);
+anyColBool = any(model.S,1);
+modelOut.rxns = model.rxns(anyColBool);
+modelOut.lb = model.lb(anyColBool);
+modelOut.ub = model.ub(anyColBool);
+modelOut.c = model.c(anyColBool);
+modelOut.rules = model.rules(anyColBool);
+modelOut.grRules = model.grRules(anyColBool);
+modelOut.grRules = model.grRules(anyColBool);
+modelOut.rxnGeneMat = model.rxnGeneMat(anyColBool,:);
+modelOut.mets = model.mets(anyRowBool);
+modelOut.b = model.b(anyRowBool);
+modelOut.S = model.S(anyRowBool,anyColBool);
+% selMets = model.mets(any(sum(abs(model.S), 2) == 0, 2));
+% if ~isempty(selMets)
+%     model = removeMetabolites(model, selMets, false);
+% end
+% modelOut = model;
+
+end
+
+
+function model = convergeIsoforms(model, separate_isoform, printLevel)
+% Some COBRA models have isoforms instead of genes. Most of the isoforms
```

Comment:
Whats used are transcripts (at least the example you give is referring to different transcripts. Thats not necessarily an isoform (Some transcripts do have different functionality).

Reply:
We have use the concept of isoform due to the mention in this paper 
"Although Recon1 (Duarte et al., 2007) was the only human metabolic model that introduced isoform-level annotations for some genes through bibliographic research, it adopted a custom annotation for isoforms, and therefore did not allow any mapping to known identifiers from public databases. For these reasons, such annotations have been subsequently lost and then simply ignored, mapping transcriptomic data with gene-level resolution only (Ryu et al., 2015). "
https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btx562/4107935

In addition, we can change the name to convergeTranscripts. The main concept is that several knockout techniques make not possible to transcribe any of the transcripts of a given gene. We have offered this oportunity for the gMCS.

Reply:
Well... Sometimes I hate simplifications (like the one done here splice-isoform = isoform and different transcripts = different isoforms). While its most often true that different transcripts have similar (or identical) function, the opposite is definitely not true (i.e. not all isoforms are encoed by the same gene).

In general, I think this is fine, but yes, renaming it would make sense to avoid confusion (also in the parameters).